jack-test/test/java/com/rapleaf/jack/test_project/database_1/impl/BaseUserPersistenceImpl.java

```diff
@@ -64,75 +56,129 @@ public User create(Map<Enum, Object> fieldsMap) throws IOException {
   }
 
   public User create(final String handle, final Long created_at_millis, final int num_posts, final Long some_date, final Long some_datetime, final String bio, final byte[] some_binary, final Double some_float, final Double some_decimal, final Boolean some_boolean) throws IOException {
-    long __id = realCreate(new AttrSetter() {
-      public void set(PreparedStatement stmt) throws SQLException {
-          stmt.setString(1, handle);
-        if (created_at_millis == null) {
-          stmt.setNull(2, java.sql.Types.INTEGER);
-        } else {
-          stmt.setLong(2, created_at_millis);
+    StatementCreator statementCreator = new StatementCreator() {
+      private final List<String> nonNullFields = new ArrayList<>();
+      private final List<AttrSetter> statementSetters = new ArrayList<>();
+
+      {
+        int index = 1;
+
+        nonNullFields.add("handle");
+        int fieldIndex0 = index++;
+        statementSetters.add(stmt -> stmt.setString(fieldIndex0, handle));
+
+        if (created_at_millis != null) {
+          nonNullFields.add("created_at_millis");
+          int fieldIndex1 = index++;
+          statementSetters.add(stmt -> stmt.setLong(fieldIndex1, created_at_millis));
         }
-          stmt.setInt(3, num_posts);
-        if (some_date == null) {
-          stmt.setNull(4, java.sql.Types.DATE);
-        } else {
-          stmt.setDate(4, new Date(some_date));
+
+        nonNullFields.add("num_posts");
+        int fieldIndex2 = index++;
+        statementSetters.add(stmt -> stmt.setInt(fieldIndex2, num_posts));
```

Comment:
Using a different variable when setting the index is necessary because variables in lambda must be final.